src/AppRoutes.tsx

```diff
@@ -33,7 +38,30 @@ export const AppRoutes = () => (
 
     {/* special pages */}
     <Route path="admin/variables" element={<VariablesPage />} />
-    <Route path="admin/administration" element={<AdministrationPage />} />
+    <Route
+      path="admin/administration"
+      element={
+        <ProtectedRoute role={AUTH_ROLE.Admin}>
+          <AdministrationPage />
+        </ProtectedRoute>
+      }
+    />
     <Route path="*" element={<NotFoundPage />} />
   </Routes>
 );
+
+export const ProtectedRoute = ({ children, role }: React.PropsWithChildren<IProtectedRouteProps>) => {
+  if (keycloakService.isAuthenticated()) {
+    if (keycloakService.hasRealmRole(role)) {
+      return <>{children}</>;
+    } else {
+      return <div>User not allowed</div>;
+    }
+  } else {
+    keycloakService.login().catch(() => {
+      throw new Error('Keycloak login failed.');
+    });
+  }
+
+  return <NotFoundPage />;
```

Comment:
I think some error should be reported instead of displaying `NotFoundPage`? Let's look at it during demo meeting.